### PR TITLE
`BannerLayout`

### PR DESCRIPTION
**Pull request by imashnake0**
[comment]: # (Replace [ ] with [x].)
- [x] Read [`CONTRIBUTING.md`](https://github.com/imashnake0/Animite/blob/be53ba4561c8ff20f588fb348965f208e301b6b8/CONTRIBUTING.md).

New reusable `BannerLayout`.

**Summary of changes:**

1. Create `BannerLayout`.

**Tested changes:**

`HomeScreen` and `MediaPage` are unchanged.

[comment]: # (THANK YOU! ♡\(灬♥ ◡ ♥灬\))

### CODE REVIEW

File: app/src/main/java/com/imashnake/animite/features/home/HomeScreen.kt [hunk 1]
```diff
@@ -87,88 +87,76 @@ fun HomeScreen(
...
-                            contentDescription = null,
-                            modifier = Modifier
-                                .fillMaxWidth()
-                                .height(dimensionResource(R.dimen.banner_height))
-                                .bannerParallax(scrollState),
-                            contentScale = ContentScale.Crop,
-                            alignment = Alignment.TopCenter
-                        )
+                    BannerLayout(
+                        banner = { bannerModifier ->
```
Comment (boswelja) [inline app/src/main/java/com/imashnake/animite/features/home/HomeScreen.kt:97]: Can we avoid this "please use the modifier" pattern somehow? We've got a few cases at work and it's not fun

File: app/src/main/java/com/imashnake/animite/features/media/MediaPage.kt [hunk 2]
```diff
@@ -90,124 +91,120 @@ fun MediaPage(
...
-                    )
-
-                    if (!media.ranks.isNullOrEmpty()) {
-                        MediaRankings(
-                            rankings = media.ranks,
+                BannerLayout(
+                    banner = { bannerModifier ->
+                        MediaBanner(
+                            imageUrl = media.bannerImage,
+                            tintColor = Color(media.color ?: 0).copy(alpha = 0.25f),
```
Comment (boswelja) [inline app/src/main/java/com/imashnake/animite/features/media/MediaPage.kt:111]: Can we have `media` fall back to black or something before it makes its way to the UI? Non-nullable color would be very convenient

File: core/src/main/kotlin/com/imashnake/animite/core/ui/layouts/BannerLayout.kt [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.unit.Dp
+import com.imashnake.animite.core.R
+import com.imashnake.animite.core.ui.LocalPaddings
+
+@Composable
+fun BannerLayout(
```
Comment (boswelja) [inline core/src/main/kotlin/com/imashnake/animite/core/ui/layouts/BannerLayout.kt:21]: Where docs

File: app/src/main/java/com/imashnake/animite/features/media/MediaPage.kt (hunk 2 above)
Comment (imashnake0) [reply app/src/main/java/com/imashnake/animite/features/media/MediaPage.kt:111]: https://github.com/imashnake0/Animite/blob/6e3eda11885b23bed2a81f2028647f57e2762130/app/src/main/java/com/imashnake/animite/features/media/MediaPageViewModel.kt#L33 will be fixed in a different PR

File: app/src/main/java/com/imashnake/animite/features/home/HomeScreen.kt (hunk 1 above)
Comment (imashnake0) [reply app/src/main/java/com/imashnake/animite/features/home/HomeScreen.kt:97]: Documented :)

**Review by boswelja — APPROVED** [review]
Close enough

**Pull request merged**